[bug] 3.6.0-dev Timestamp parse errors

@fukusuket Could you take a look at this when you get time? I confirmed that it does not happen on 3.4.0 and 3.5.0. Only on the `main` branch 3.6.0-dev with the latest crates. I think something broke after running `cargo update`.

Command: `./target/release/hayabusa csv-timeline -d ../hayabusa-sample-evtx -w -o sample.csv`

It gives `Errors were generated. Please check ./logs/errorlog-20250915_100738.log for details.`

with lots of these errors:
```
[ERROR] Timestamp parse error.
Input: 2020-07-12T05:10:08.442192Z
Error: input contains invalid characters

[ERROR] Timestamp parse error.
Input: 2020-07-12T19:45:00.670312Z
Error: input contains invalid characters

[ERROR] Timestamp parse error.
Input: 2020-07-12T19:45:00.670312Z
Error: input contains invalid characters
```

--- a/src/main.rs
+++ b/src/main.rs
@@ -2450,7 +2450,7 @@ Any hostnames added to the critical_systems.txt file will have all alerts above 
         } else if is_splunk_api_json {
             "%Y-%m-%dT%H:%M:%S%.9fZ"
         } else {
-            "%Y-%m-%dT%H:%M:%S%.3fZ"
+            "%Y-%m-%dT%H:%M:%S%.fZ"
         };
         // EventID側の条件との条件の混同を防ぐため時間でのフィルタリングの条件分岐を分離した
         let timestamp = match NaiveDateTime::parse_from_str(
@@ -2578,7 +2578,7 @@ Any hostnames added to the critical_systems.txt file will have all alerts above 
                                     without_timezone_datetime,
                                     Utc,
                                 )
-                                .format("%Y-%m-%dT%H:%M:%S%.3fZ")
+                                .format("%Y-%m-%dT%H:%M:%S%.fZ")
                             }
                             Err(e) => {
                                 AlertMessage::warn(&format!(
@@ -2591,7 +2591,7 @@ Any hostnames added to the critical_systems.txt file will have all alerts above 
                                     e
                                 ))
                                 .ok();
-                                DateTime::<Utc>::default().format("%Y-%m-%dT%H:%M:%S%.3fZ")
+                                DateTime::<Utc>::default().format("%Y-%m-%dT%H:%M:%S%.fZ")
                             }
                         };
 

--- a/src/timeline/log_metrics.rs
+++ b/src/timeline/log_metrics.rs
@@ -34,44 +34,42 @@ impl LogMetrics {
             )
             .map(|evt_value| evt_value.to_string().replace("\\\"", "").replace('"', ""))
             {
-                let timestamp =
-                    match NaiveDateTime::parse_from_str(evttime.as_str(), "%Y-%m-%dT%H:%M:%S%.3fZ")
-                    {
-                        Ok(without_timezone_datetime) => {
-                            Some(DateTime::<Utc>::from_naive_utc_and_offset(
-                                without_timezone_datetime,
-                                Utc,
-                            ))
-                        }
-                        Err(_) => {
-                            match NaiveDateTime::parse_from_str(
-                                evttime.as_str(),
-                                "%Y-%m-%dT%H:%M:%S%.3f%:z",
-                            ) {
-                                Ok(splunk_json_datetime) => {
-                                    Some(DateTime::<Utc>::from_naive_utc_and_offset(
-                                        splunk_json_datetime,
-                                        Utc,
-                                    ))
+                let timestamp = match NaiveDateTime::parse_from_str(
+                    evttime.as_str(),
+                    "%Y-%m-%dT%H:%M:%S%.fZ",
+                ) {
+                    Ok(without_timezone_datetime) => Some(
+                        DateTime::<Utc>::from_naive_utc_and_offset(without_timezone_datetime, Utc),
+                    ),
+                    Err(_) => {
+                        match NaiveDateTime::parse_from_str(
+                            evttime.as_str(),
+                            "%Y-%m-%dT%H:%M:%S%.3f%:z",
+                        ) {
+                            Ok(splunk_json_datetime) => {
+                                Some(DateTime::<Utc>::from_naive_utc_and_offset(
+                                    splunk_json_datetime,
+                                    Utc,
+                                ))
+                            }
+                            Err(e) => {
+                                let errmsg = format!(
+                                    "Timestamp parse error.\nInput: {evttime}\nError: {e}\n"
+                                );
+                                if stored_static.verbose_flag {
+                                    AlertMessage::alert(&errmsg).ok();
                                 }
-                                Err(e) => {
-                                    let errmsg = format!(
-                                        "Timestamp parse error.\nInput: {evttime}\nError: {e}\n"
-                                    );
-                                    if stored_static.verbose_flag {
-                                        AlertMessage::alert(&errmsg).ok();
-                                    }
-                                    if !stored_static.quiet_errors_flag {
-                                        ERROR_LOG_STACK
-                                            .lock()
-                                            .unwrap()
-                                            .push(format!("[ERROR] {errmsg}"));
-                                    }
-                                    None
+                                if !stored_static.quiet_errors_flag {
+                                    ERROR_LOG_STACK
+                                        .lock()
+                                        .unwrap()
+                                        .push(format!("[ERROR] {errmsg}"));
                                 }
+                                None
                             }
                         }
-                    };
+                    }
+                };
                 if let Some(timestamp) = timestamp {
                     if self.first_timestamp.is_none() || timestamp < self.first_timestamp.unwrap() {
                         self.first_timestamp = Some(timestamp);

--- a/src/timeline/metrics.rs
+++ b/src/timeline/metrics.rs
@@ -132,7 +132,7 @@ impl EventMetrics {
             .unwrap();
         let evtx_service_released_date = Some(DateTime::<Utc>::from_naive_utc_and_offset(dt, Utc));
         let mut check_start_end_time = |evttime: &str| {
-            let timestamp = match NaiveDateTime::parse_from_str(evttime, "%Y-%m-%dT%H:%M:%S%.3fZ") {
+            let timestamp = match NaiveDateTime::parse_from_str(evttime, "%Y-%m-%dT%H:%M:%S%.fZ") {
                 Ok(without_timezone_datetime) => Some(DateTime::<Utc>::from_naive_utc_and_offset(
                     without_timezone_datetime,
                     Utc,

--- a/src/timeline/timelines.rs
+++ b/src/timeline/timelines.rs
@@ -910,7 +910,7 @@ mod tests {
         assert_eq!(
             timeline.stats.start_time,
             Some(DateTime::<Utc>::from_naive_utc_and_offset(
-                NaiveDateTime::parse_from_str("2021-12-23T00:00:00.000Z", "%Y-%m-%dT%H:%M:%S%.3fZ")
+                NaiveDateTime::parse_from_str("2021-12-23T00:00:00.000Z", "%Y-%m-%dT%H:%M:%S%.fZ")
                     .unwrap(),
                 Utc
             ))
@@ -918,7 +918,7 @@ mod tests {
         assert_eq!(
             timeline.stats.end_time,
             Some(DateTime::<Utc>::from_naive_utc_and_offset(
-                NaiveDateTime::parse_from_str("2022-12-23T00:00:00.000Z", "%Y-%m-%dT%H:%M:%S%.3fZ")
+                NaiveDateTime::parse_from_str("2022-12-23T00:00:00.000Z", "%Y-%m-%dT%H:%M:%S%.fZ")
                     .unwrap(),
                 Utc
             ))

